internal/driver/glfw/window.go

```diff
@@ -430,17 +430,29 @@ func (w *window) Close() {
 		return
 	}
 
-	w.viewLock.Lock()
-	w.closing = true
-	w.viewLock.Unlock()
-	w.viewport.SetShouldClose(true)
+	// set w.closing flag inside draw thread to ensure we can free textures
+	runOnDraw(w, func() {
+		w.viewLock.Lock()
+		w.closing = true
+		w.viewLock.Unlock()
+		w.viewport.SetShouldClose(true)
+		cache.RangeTexturesFor(w.canvas, func(obj fyne.CanvasObject) {
+			w.canvas.Painter().Free(obj)
+		})
+	})
 
+	// destroy current renderers
 	w.canvas.WalkTrees(nil, func(node *common.RenderCacheNode) {
 		if wid, ok := node.Obj().(fyne.Widget); ok {
 			cache.DestroyRenderer(wid)
 		}
 	})
 
+	// remove all canvas objects from the cache
+	w.QueueEvent(func() {
+		cache.ForceCleanFor(w.canvas)
```

Comment:
This is not an input event, why is it being queued with them?

Reply:
Because we need to ensure that the objects are removed when there is no more input events that depend on them running, and a way I think we could accomplish it is adding it to the event queue

Reply:
Perhaps this is too soon then? As the window is closing we will at some point know that there are no pending events.
Maybe in the actual destroy instead of the `Close()` API call?

Reply:
> Maybe in the actual destroy instead of the Close() API call?

Yes that is a possibility, however we are going to do cleaning tasks in two different functions and it will not be consistent with gomobile driver that does not have that function. IMHO, calling this here does not represent any problem, it keeps consistency between the two drivers and the code for clean the cache on Close is centralized.

Reply:
Calling it here does not represent a problem. Adding it to the user input event queue is confusing.
If it can be called without this workaround in another place (such as the GLFW driver's code that actually destroys windows) it will be cleaner within this driver.
I don't think that keeping the drivers "consistent" in code placement necessarily makes sense with regards to windows as they are not actually following the same lifecycle internally.

Reply:
Got it. Done in 569a354